### PR TITLE
impl check_error_condition

### PR DESCRIPTION
**Pull request by Jemiiah**
- [✅] issue #229 
- [✅] follows contribution [guide](https://github.com/keep-starknet-strange/shinigami/blob/main/CONTRIBUTING.md)
- [ ] code change includes tests

### CODE REVIEW

**Comment by Jemiiah** [comment]
Hello ser sorry for the late reply will work on it asap

**Comment by Jemiiah** [comment]
update ser

**Comment by b-j-roberts** [comment]
I'm getting a ton of failed tests after using `scarb test` on this change

File: packages/engine/src/engine.cairo [hunk 1]
```diff
@@ -203,6 +205,43 @@ pub impl EngineExtrasImpl<T, +Drop<T>> of EngineExtrasTrait<T> {
...
+            return Result::Err(Error::SCRIPT_NON_CLEAN_STACK);
+        }
+
+        // Check if stack has at least one item
+        if self.dstack.len() < 1 {
+            return Result::Err(Error::SCRIPT_EMPTY_STACK);
+        }
+
+        // Check the final stack value
+        match self.dstack.pop_bool() {
```
Comment (b-j-roberts) [inline packages/engine/src/engine.cairo:233]: I think this pop_book instead of peeking the top stack element might be causing a lot of the test failures.

**Review by b-j-roberts — APPROVED** [review]

**Review by b-j-roberts — APPROVED** [review]
Went ahead and cleaned up those failing tests, because I wanted to have this change merged for a refactor I am working on. Thank you!

**Pull request merged**